pennylane_lightning/lightning_qubit/lightning_qubit.py

```diff
@@ -235,6 +238,26 @@ def accepted_observables(obs: qml.operation.Operator) -> bool:
     return obs.name in _observables
 
 
+def adjoint_observables(obs: qml.operation.Operator) -> bool:
```

Comment:
Is the function still necessary after the introduction of `_assert_adjdiff_no_projectors`?

Reply:
`_assert_adjdiff_no_projectors` is inside `LightningBase`. My intention was to leave that as is, and add this function here which can be reused for kokkos and GPU once those are ported over to use the new API.